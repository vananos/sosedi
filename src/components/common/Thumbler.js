import React, { Component } from 'react'
import "../Thumbler.css";
export default class Thumbler extends Component {
  render() {
    const {value1, value2, name} = props;
    return (
      <div>
        <input type="radio" value={value1} name={name}>       
        <input type="radio" value={value2} name={name}>       
      </div>
    )
  }
}
