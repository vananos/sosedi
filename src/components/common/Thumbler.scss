@import "../../base.scss";

.thumbler {
    position: relative;
    width: 100%;
    height: 100px;
    background: url("") no-repeat;
    background-size: 25% 25%;
    background-position: bottom 10px right 10px;
    background-color: $light-grey;

    .yes-btn,
    .no-btn {
        position: absolute;
        background-image: url("../../assets/common/thumb-up.svg");
        background-size: 25px;
        background-repeat: no-repeat;
        width: 33%;
        height: 100%;
        padding: 0;
        margin: 0;
        border: none;
    }

    .no-btn {
        top: 0;
        left: 0;
        background-position: right 10px top 10px;
        transform: rotate(180deg);
    }

    .yes-btn {
        top: 0;
        right: 0;
        background-position: right 10px bottom 10px;
    }

    .target {
        background-color: $light-brown;
        box-sizing: border-box;
        border: 10px solid $light-brown;
        background: $brown;
        width: 33%;
        height: 100%;
        margin: auto;
        transition: 0.5s ease-in;
        display: flex;
        justify-content: center;
        align-items: center;
    }

    .target.yes {
        margin-right: 0;
        border: 10px solid $green-yes;
        background: $green-yes;
        transition: 0.5s ease-in;
    }

    .target.no {
        margin-left: 0;
        border: 10px solid $red-no;
        background: $red-no;
        transition: 0.5s ease-in;
    }
}