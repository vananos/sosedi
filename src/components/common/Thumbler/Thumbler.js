import React, { Component } from "react";
import "./Thumbler.scss";
import PropTypes from "prop-types";

export default class Thumbler extends Component {
  constructor(props) {
    super(props);
    this.state = {
      targetState: props.targetState || "default"
    };
  }

  updateState = (e, state) => {
    this.setState({ targetState: state });
    e.stopPropagation();
  };

  render() {
    const { children, name } = this.props;
    const targetState = this.state.targetState;
    return (
      <div className="thumbler" onClick={e => this.updateState(e, "default")}>
        <input type="text" name={name} hidden value={targetState} readOnly/>
        <div
          className={`no-btn ${targetState === "no" ? "no" : ""}`}
          onClick={e => this.updateState(e, "no")}
        />
        <div className={`target ${targetState}`}>{children}</div>
        <div
          className={`yes-btn ${targetState === "yes" ? "yes" : ""}`}
          onClick={e => this.updateState(e, "yes")}
        />
      </div>
    );
  }
}

Thumbler.propTypes = {
  children: PropTypes.node.isRequired
};
