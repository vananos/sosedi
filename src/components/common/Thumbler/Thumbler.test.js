import { shallow } from "enzyme";
import React from "react";
import Thumbler from "./Thumbler";

describe("Thumbler, ", () => {
  let thumblerWrapper;

  beforeEach(() => {
    thumblerWrapper = shallow(
      <Thumbler>
        <span>Like?</span>
      </Thumbler>
    );
  });

  it("default state", () => {
    expect(thumblerWrapper).toMatchSnapshot();
  });

  it("click on yes -> thumbler should be in `yes` state", () => {
    thumblerWrapper
      .find(".yes-btn")
      .first()
      .simulate("click", { stopPropagation() {} });

    expect(thumblerWrapper).toMatchSnapshot();
  });

  it("click on no -> thumbler should be in `no` state", () => {
    thumblerWrapper
      .find(".no-btn")
      .first()
      .simulate("click", { stopPropagation() {} });

    expect(thumblerWrapper).toMatchSnapshot();
  });
});
