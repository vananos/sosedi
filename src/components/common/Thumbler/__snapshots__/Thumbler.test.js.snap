// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thumbler,  click on no -> thumbler should be in \`no\` state 1`] = `
<div
  className="thumbler"
  onClick={[Function]}
>
  <div
    className="no-btn no"
    onClick={[Function]}
  />
  <div
    className="target no"
  >
    <span>
      Like?
    </span>
  </div>
  <div
    className="yes-btn "
    onClick={[Function]}
  />
</div>
`;

exports[`Thumbler,  click on yes -> thumbler should be in \`yes\` state 1`] = `
<div
  className="thumbler"
  onClick={[Function]}
>
  <div
    className="no-btn "
    onClick={[Function]}
  />
  <div
    className="target yes"
  >
    <span>
      Like?
    </span>
  </div>
  <div
    className="yes-btn yes"
    onClick={[Function]}
  />
</div>
`;

exports[`Thumbler,  default state 1`] = `
<div
  className="thumbler"
  onClick={[Function]}
>
  <div
    className="no-btn "
    onClick={[Function]}
  />
  <div
    className="target default"
  >
    <span>
      Like?
    </span>
  </div>
  <div
    className="yes-btn "
    onClick={[Function]}
  />
</div>
`;
